docs/DeveloperGuide.md

```diff
@@ -1,34 +1,1260 @@
+
+### 3.3. Commands Component (Neil)
+
+Every command that a user can input into **Zer0Note** is represented by an object that extends the abstract class
+`CliCommand`. `CliCommand` contains some basic fields and methods that are shared by all types of commands, such as:
+
+* `execute()`, which is called after setting up the object appropriately, to perform the action requested by the user;
+
+* `appState`, which refers to the `AppState` object that stores the state of the currently running instance of
+**Zer0Note**; and
+
+* `commandParams`, which is a String containing the parameters the user entered as parameters to the command.
+
+The following 2 UML diagrams show the different types of `CliCommand` components used in the application.
+
+This diagram describes the `CliCommand`s related to the Timetable mode.
+
+![UML diagram for Timetable Commands](diagrams/class/jpeg/timetable_commands.jpg)
+
+And this diagram describes the `CliCommand`s related to the Notebook mode.
+
+![UML diagrams for Notebook Commands](diagrams/class/jpeg/notebook_commands.jpg)
+
+// TODO: add the new commands for "Find", "Tag", "SortByDate".
+
+### 3.4. Tasks Component
+
+![UML Diagram from Task Component](diagrams/class/jpeg/taskComponent.jpg)
+
+Figure []. Structure of Tasks Component
+
+The `Tasks` component,
+
+- stores an array of `Task` objects that represents the deadline of the user.
+- together with the `Notebooks` component form the `AppState` component.   
+
+The `TaskList` class,
+
+- has methods to add a new `Task` object and remove existing one at an index.
+
+### 3.5. Notebooks Component
+
+![UML diagram for Notebooks Component](diagrams/class/jpeg/notebooks_simplified.jpg)
+
+Figure []. Structure of Notebooks Component
+
+The `Notebooks` class,
+
+- stores a `NotebookShelf` object that contains a list of `Notebook` objects.
+- together with the `Tasks` component form the `AppState` component.
+
+The `NotebookShelf` class,
+
+- has methods to add new `Notebook` object and remove existing one.
+- has methods to search through to find a `Notebook` object with matching titles.
+
+The `Notebook` class,
+
+- contains a title and a list of `Section` objects.
+- has methods to add new `Section` object or remove existing one.
+- has a `tag` field that user can be set and get.
+
+The `Section` object,
+
+- contains a title and list of `Page` objects.
+- has methods to add new `Page` object or remove existing one.
+- has a `tag` field that user can be set and get.
+
+The `Page` object,
+
+- contains a title, and the content of the page as a String object.
+- has a `tag` field that user can be set and get.
+
+### 3.6. Storage Component
+
+![UML diagram for Storage](diagrams/class/jpeg/Storage_UML_class.jpg)
+
+The `Storage` component,
+
+* Contains the method `saveToFile` to save the current AppState of the application in the `notebooks.txt` and `tasks.txt` files.  
```

Comment:
Is the txt file stored in the same directory or is it stored in another directory?